File: content/examples/other/leetcode/problems/3sum/example-1/solution.js
function threeSumBrute(nums) {
  const triples = new Set();
  const n = nums.length;
  for (let i = 0; i < n; i += 1) {
    for (let j = i + 1; j < n; j += 1) {
      for (let k = j + 1; k < n; k += 1) {
        if (nums[i] + nums[j] + nums[k] === 0) {
          const t = [nums[i], nums[j], nums[k]].sort((a, b) => a - b);
          triples.add(t.join(","));
        }
      }
    }
  }
  return Array.from(triples, (s) => s.split(",").map((x) => Number(x)));
}

if (require.main === module) {
  console.log(threeSumBrute([-1, 0, 1, 2, -1, -4]));
}

module.exports = { threeSumBrute };

File: content/examples/other/leetcode/problems/3sum/example-1/solution.py
def three_sum_brute(nums):
    triples = set()
    n = len(nums)
    for i in range(n):
        for j in range(i + 1, n):
            for k in range(j + 1, n):
                if nums[i] + nums[j] + nums[k] == 0:
                    t = tuple(sorted((nums[i], nums[j], nums[k])))
                    triples.add(t)
    return [list(t) for t in triples]


if __name__ == "__main__":
    print(three_sum_brute([-1, 0, 1, 2, -1, -4]))
